ui/src/test/java/edu/wpi/grip/ui/codegeneration/BlurGenerationTest.java

```diff
@@ -49,13 +49,37 @@ void generatePipeline() {
 			else if(sock.getSocketHint().isCompatibleWith(Outputs.createNumberSocketHint("Number", blurRatio))){
 				sock.setValue(blurRatio);
 			}
+			else if(sock.getSocketHint().getIdentifier().equals("Type")){
+				Object[] options = (Object[])sock.getSocketHint().getDomain().get();
+				for(Object option: options){
+					if(option.toString().equals(blurType)){
+						sock.setValue(option);
+					}
+				}
+			}
 		}
 	}
 	@Test
-	public void BoxBlurTest(){
-		test( () -> {generatePipeline(); return true;},
+	public void boxBlurTest(){
+		test( () -> {generatePipeline("Box Blur"); return true;},
```

Comment:
This along with several other tests would be better as parameterized tests. 
However as seen in [Junit issue #751](https://github.com/junit-team/junit4/issues/751) Category tests do not work with parameterized tests. The category is used to run just the generation tests instead of all of the UI tests, useful when developing new generation patterns and if we decide to move generation tests to separate test systems.